src/main/resources/templates/challenge.html

```diff
@@ -3,30 +3,30 @@
 <body>
 <div class="container" layout:fragment="content">
     <!--/*@thymesVar id="challenge" type="org.owasp.wrongsecrets.challenges.ChallengeUI"*/-->
-    <p class="h1 mt-2"> <span th:text="${challenge.name}"/> <span th:block th:text="${challenge.stars}"></span></p>
+    <p class="h1 mt-2"> <span th:text="${challenge.name}" th:attr="data-cy=challenge-title"/> <span th:block th:text="${challenge.stars}"></span></p>
```

Comment:
love this! this is actually a nice accessibility upgrade! thank you! (great side-effect ;-) )